test/src/test/java/org/corfudb/integration/LogReplicationLogicalGroupIT.java

```diff
@@ -0,0 +1,728 @@
+            }
+        }
+    }
+
+    /**
+     * This test has the similar workflow as testSnapshotAndLogEntrySync() in
+     * {@link LogReplicationDynamicStreamIT}, with Logical Group feature.
+     *
+     * This IT tests the correctness of log replication if
+     * only source tables(srcFederatedTables, srcTablesGroupA,
+     * srcTablesGroupB) are opened before log replication.
+     */
+    @Test
+    public void testSourceNewTablesReplication() throws Exception {
+        // Register client and add group destinations
+        CorfuRuntime clientRuntime = getClientRuntime();
+        LogReplicationLogicalGroupClient logicalGroupClient =
+                new LogReplicationLogicalGroupClient(clientRuntime, SAMPLE_CLIENT_NAME);
+        logicalGroupClient.addDestination(GROUP_A, DefaultClusterConfig.getSinkClusterIds().get(SINK2_INDEX));
+        logicalGroupClient.addDestination(GROUP_B, DefaultClusterConfig.getSinkClusterIds().get(SINK3_INDEX));
+
+        // Open tables for replication on Source side
+        int numTables = 2;
+        openFederatedTable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcFederatedTables);
+        openGroupATable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcTablesGroupA);
+        openGroupBTable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcTablesGroupB);
+
+        // Write data to tables opened on each source cluster
+        writeDataOnSource(0, NUM_WRITES);
+
+        // Start the replication
+        startReplicationServers();
+
+        // Verify all the sessions' snapshot sync completed
+        verifyInLogEntrySyncState(SINK1_INDEX, SessionManager.getDefaultSubscriber());
+        verifyInLogEntrySyncState(SINK2_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+        verifyInLogEntrySyncState(SINK3_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+
+        // Open tables on sink
+        openFederatedTable(numTables, sinkCorfuStores.get(SINK1_INDEX), sinkFederatedTables);
+        openGroupATable(numTables, sinkCorfuStores.get(SINK2_INDEX), sinkTablesGroupA);
+        openGroupBTable(numTables, sinkCorfuStores.get(SINK3_INDEX), sinkTablesGroupB);
+
+        // Verify the data is in table on the sink side
+        verifyFederatedTableDataOnSink(sinkCorfuStores.get(SINK1_INDEX), NUM_WRITES, sinkFederatedTables);
+        verifyGroupATableDataOnSink(sinkCorfuStores.get(SINK2_INDEX), NUM_WRITES, sinkTablesGroupA);
+        verifyGroupBTableDataOnSink(sinkCorfuStores.get(SINK3_INDEX), NUM_WRITES, sinkTablesGroupB);
+
+        // Write data on sources
+        writeDataOnSource(NUM_WRITES, NUM_WRITES);
+
+        // Verify all the sessions' snapshot sync completed
+        verifyInLogEntrySyncState(SINK1_INDEX, SessionManager.getDefaultSubscriber());
+        verifyInLogEntrySyncState(SINK2_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+        verifyInLogEntrySyncState(SINK3_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
```

Comment:
Why snapshot sync completed here? they all went into log entry sync before you open tables on Sink side right? the following writes are for log entry sync

Reply:
Yeah the following writes are for log entry sync. Just removed those three lines of code.